### ISSUE TITLE
MyAnimeList support 

### ISSUE THREAD

**Issue by Zabooby** [hayase-app/ui#134]
Is support MAL sync planned?

**Comment by ThaUnknown** [comment]
no, MAL API is just dogshit [on the rare occasion it works], and I don't want to rely on fan-made 3rd part API's for MAL

however AniList exposes MAL ID's for most anime, so someone could easily make a PR based on that, which I'd likely accept

invalid issue template

**Comment by ThaUnknown** [comment]
I can't even look at their documentation
https://haven.pages.dev/video-player/public/?file=https://cdn.discordapp.com/attachments/631919775613845504/962704977464266762/Desktop_2022.04.10_-_15.22.55.05.mp4

**Comment by Zabooby** [comment]
Couldn't you just copy taiga's implementation of MAL,they seem to have it working?

https://github.com/erengy/taiga

**Comment by ThaUnknown** [comment]
![image](https://user-images.githubusercontent.com/6506529/162620473-1f2c2927-f6e0-401d-a478-918c75bbe6e0.png)
![image](https://user-images.githubusercontent.com/6506529/162620480-bb5d8adf-c78c-44ce-9692-987505175d06.png)
no lol

**Comment by Zabooby** [comment]
Here is some documentation on their api, might help.

https://myanimelist.net/apiconfig

**Comment by ThaUnknown** [comment]
> Here is some documentation on their api, might help.
> 
> https://myanimelist.net/apiconfig

![image](https://user-images.githubusercontent.com/6506529/162621242-935f98e3-7433-451a-82d1-2e81b88a4e44.png)

**Comment by Zabooby** [comment]
https://myanimelist.net/apiconfig/references/authorization
https://myanimelist.net/static/apiagreement.html
https://myanimelist.net/apiconfig/references/api/v2

Can you see these, maybe its because you don't have a MAL account?

**Comment by Zabooby** [comment]
Nvm I checked the video, have you tried using a different browser or I could send you a html file of the docs if you want.

**Comment by ThaUnknown** [comment]
> Nvm I checked the video, have you tried using a different browser or I could send you a html file of the docs if you want.

that video is the perfect example of why I don't bother supporting MAL, if some1 makes a PR for it, I'll accept it, but I don't plan on working on it myself

**Comment by Olyno** [comment]
Hi 👋🏻
I would add Anilist too if possible. MyAnimeList starts to be deserted more and more to go for Anilist, which is more modern in every sense. Note that the Anilist api is particularly simple, hence the reason for my proposal.

**Comment by ThaUnknown** [comment]
> Hi 👋🏻 I would add Anilist too if possible. MyAnimeList starts to be deserted more and more to go for Anilist, which is more modern in every sense. Note that the Anilist api is particularly simple, hence the reason for my proposal.

I recommend u read the list of features https://github.com/ThaUnknown/miru#features
anilist is already supported, idk how u missed the big ass "login with anilist" button in the corner

**Comment by Olyno** [comment]
Oops sorry I didn't see it, I just stumbled upon this issue by chance, I should have been more careful, my apologies

**Comment by Zabooby** [comment]
Would any of these help with adding mal support? 

https://github.com/Kylart/MalScraper
https://github.com/Chris-Kode/myanimelist-api-v2
https://github.com/jikan-me/jikan

**Comment by YUKI2eN3e** [comment]
> ![image](https://user-images.githubusercontent.com/6506529/162620473-1f2c2927-f6e0-401d-a478-918c75bbe6e0.png) ![image](https://user-images.githubusercontent.com/6506529/162620480-bb5d8adf-c78c-44ce-9692-987505175d06.png) no lol

Maybe I'm completely misunderstanding how it works (I've only looked into Wasm pretty briefly), but couldn't you compile the C++/C code to Wasm and use it with your JavaScript code that way?

**Comment by ThaUnknown** [comment]
wow i cringed

It's not that its not possible, its that i cant be fucking arsed to do it, because of how dogshit MAL is

**Comment by Muril-o** [comment]
Reopening this since its a good feature that many people would benefit from. It's not impossible, just troublesome because MAL has chosen it to be.
As ThaUnknown said, he won't work on it himself but will accept PRs if someone else is willingly to implement.

**Comment by Atkoset** [comment]
Hello, could you add Kitsu API too if possible? Kitsu is comparable to Anilist in the way that it also syncs your anime. Where it differs is that it's fully opensource option that a lot of people including me use since it's opensource it shouldn't take too much work to integrate I hope. I have included a link to their API repo below.

https://github.com/hummingbird-me/api-docs 

@ThaUnknown

**Comment by ThaUnknown** [comment]
as I stated before, I won't be pursuing this, but I'm open to accept PRs for it

**Comment by ThaUnknown** [comment]
was added a while back
